src/tests/JIT/HardwareIntrinsics/Arm/Directory.Build.targets

```diff
@@ -0,0 +1,35 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project>
+
+  <Import Project="$([MSBuild]::GetPathOfFileAbove(Directory.Build.targets, $(MSBuildThisFileDirectory)..))" />
+
+  <ItemGroup>
+    <ProjectReference Include="$(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.csproj" ReferenceOutputAssembly="false" />
+  </ItemGroup>
+
+  <PropertyGroup>
+    <GeneratedHWIntrinsicTestDirectory Condition="'$(GeneratedHWIntrinsicTestDirectory)' == ''">$(IntermediateOutputPath)$(MSBuildProjectName)/gen/</GeneratedHWIntrinsicTestDirectory>
+    <GeneratedHWIntrinsicTestListFile Condition="'$(GeneratedHWIntrinsicTestListFile)' == ''">$(GeneratedHWIntrinsicTestDirectory)GeneratedHWIntrinsicTestList.txt</GeneratedHWIntrinsicTestListFile>
+  </PropertyGroup>
+
+  <Target Name="ExecuteGenerateTestsScript"
+          Inputs="$(MSBuildAllProjects);$(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.cs"
+          Outputs="$(GeneratedHWIntrinsicTestListFile)"
+          Condition="'$(Language)' == 'C#'">
+    <Exec Command="$(DotNetCli) run --project $(RepoRoot)/src/tests/Common/GenerateHWIntrinsicTests/GenerateHWIntrinsicTests_Arm.csproj -c $(Configuration) -- $(MSBuildProjectName) $(MSBuildThisFileDirectory)Shared $(GeneratedHWIntrinsicTestDirectory) $(GeneratedHWIntrinsicTestListFile)" />
+  </Target>
+
+  <Target Name="ReadGeneratedHWIntrinsicTestListFile"
+          BeforeTargets="BeforeCompile;CoreCompile"
+          DependsOnTargets="ExecuteGenerateTestsScript"
+          Returns="$(GeneratedHWIntrinsicTestList)"
+          Condition="'$(Language)' == 'C#'">
+    <ReadLinesFromFile File="$(GeneratedHWIntrinsicTestListFile)">
+      <Output TaskParameter="Lines" ItemName="GeneratedHWIntrinsicTestList" />
+    </ReadLinesFromFile>
+    <ItemGroup>
+      <Compile Include="@(GeneratedHWIntrinsicTestList)" />
+    </ItemGroup>
+  </Target>
+
+</Project>
```

Comment:
This is the actual interesting "bit" of the change.

Each folder has a targets file that:
1. Has a dependency on the relevant test generator to ensure it is built first
2. A target that runs the test generator to ensure the necessary files exist, with relevant up to date checks to ensure it doesn't run unnecessarily
3. A target that has to always run as part of compilation to read the list of generated tests and include them in the list of files csc should process

Reply:
The actual `GenerateTests.csx` was simply moved to be part of a csproj and to take a couple inputs to help filter the tests.

It's not necessarily the cleanest change, but it is by far one of the simplest. Building a source generator or otherwise refactoring things more would result in significantly more time and churn and would block the AVX-512 work until that was completed.